alacritty_terminal/src/event_loop.rs

```diff
@@ -39,14 +36,41 @@ pub enum Msg {
     Resize(WindowSize),
 }
 
-/// The main event!.. loop.
+struct PeekableReceiver<T> {
```

Comment:
This structure seems a bit strange and inefficient to me. Did you add it solely for synchronized updates? Why was it necessary if it worked without this before?

Also it seems like you're silently ignoring channel failure. How does this impact the behavior of the channel's death compared to master?

Reply:
The channel was previously registered into the event loop, while this version isn't. I added a panic on channel death.

Reply:
Could you move it and related impl the bottom of the file.